src/mem/largealloc.h

```diff
@@ -58,7 +58,7 @@ namespace snmalloc
   class MemoryProviderStateMixin : public PAL
   {
     std::atomic_flag lock = ATOMIC_FLAG_INIT;
-    address_t bump;
+    char* bump;

```

Comment:
Why `char*` and not `void*`?  This means we need some explicit casts later and are doing explicit pointer arithmetic on `char*` rather than with the `pointer_offset` function, which makes auditing the code for pointer arithmetic harder.

Reply:
I think this change predates the existence of `pointer_offset`; I've moved things over to `void *` and `pointer_offset`, though there's still pointer subtraction without a wrapper method.